### PR TITLE
Bluetooth: Controller: Fix scan window value corruption

### PR DESCRIPTION
**Pull request by cvinayak**
Fix scan window value corruption due to modification of stored ticks_window value when enabling continuous scan.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

File: subsys/bluetooth/controller/ll_sw/ull_scan.c [hunk 1]
```diff
@@ -349,8 +350,9 @@ int ull_scan_reset(void)
 	return 0;
 }
 
-void ull_scan_params_set(struct lll_scan *lll, uint8_t type, uint16_t interval,
-			 uint16_t window, uint8_t filter_policy)
+uint32_t ull_scan_params_set(struct lll_scan *lll, uint8_t type,
+			     uint16_t interval, uint16_t window,
+			     uint8_t filter_policy)
```
Comment (Thalley) [inline subsys/bluetooth/controller/ll_sw/ull_scan.c:355]: Just FYI, the TSC changed the character limit to 100, removing the 80 recommendation

File: tests/bluetooth/bsim_bt/bsim_test_mesh/overlay_low_lat.conf [hunk 2]
```diff
@@ -1,4 +1,3 @@
-CONFIG_BT_EXT_ADV=n
```
Comment (Thalley) [inline tests/bluetooth/bsim_bt/bsim_test_mesh/overlay_low_lat.conf:1]: Why is extended advertising changed here?

Comment (cvinayak) [reply tests/bluetooth/bsim_bt/bsim_test_mesh/overlay_low_lat.conf:1]: `CONFIG_BT_EXT_ADV` is default `n` and an explicit `n`  in `overlay_low_lat.conf` is inconsistent with tests not using this overlay.

Comment (PavelVPV) [reply tests/bluetooth/bsim_bt/bsim_test_mesh/overlay_low_lat.conf:1]: It is y by default for mesh: https://github.com/zephyrproject-rtos/zephyr/blob/928e5bbcf08ccfe75f264d407e16fb1c7e82846e/subsys/bluetooth/Kconfig.adv#L10

Comment (cvinayak) [reply tests/bluetooth/bsim_bt/bsim_test_mesh/overlay_low_lat.conf:1]: @PavelVPV  Any reason why it is `n` for nRF51 (7367980) ? If it is only used for multiple advertising sets, then it should work on nRF51 too, did it fail on target?

Comment (PavelVPV) [reply tests/bluetooth/bsim_bt/bsim_test_mesh/overlay_low_lat.conf:1]: IIRC it didn't fit in flash or RAM.

Comment (cvinayak) [reply tests/bluetooth/bsim_bt/bsim_test_mesh/overlay_low_lat.conf:1]: ok... I will go back to `CONFIG_BT_EXT_ADV=n` for nrf51 (low lat variant).  BTW, the `periodic_adv` sample builds, runs, is extended advertising and is discoverable on iPhone 13 Pro.

File: tests/bluetooth/bsim_bt/bsim_test_mesh/prj.conf [hunk 3]
```diff
@@ -4,14 +4,18 @@ CONFIG_SYS_CLOCK_TICKS_PER_SEC=32768
...
 CONFIG_BT_DEVICE_NAME="Mesh test"
+CONFIG_BT_DEBUG_LOG=y
+
+# Bluetooth Controller
 CONFIG_BT_LL_SW_SPLIT=y
-CONFIG_BT_OBSERVER=y
-CONFIG_BT_BROADCASTER=y
-CONFIG_BT_TINYCRYPT_ECC=y
+CONFIG_BT_CTLR_ADVANCED_FEATURES=y
+CONFIG_BT_CTLR_SCAN_UNRESERVED=y
```
Comment (PavelVPV) [inline tests/bluetooth/bsim_bt/bsim_test_mesh/prj.conf:18]: What is the purpose of explicitly enabling `CONFIG_BT_CTLR_SCAN_UNRESERVED` here? It is already enabled by default for mesh: https://github.com/zephyrproject-rtos/zephyr/blob/cccfc8fc8f627e127cf81b4960c3d2419a477e19/subsys/bluetooth/controller/Kconfig.ll_sw_split#L793-L796.

**Review by PavelVPV — CHANGES_REQUESTED** [review]
Few comments from my side should be addressed.

**Review by PavelVPV — DISMISSED** [review]

**Review by Thalley — DISMISSED** [review]

**Comment by cvinayak** [comment]
@PavelVPV and @Thalley This fix causes the friendship msg_mesh_low_lat bsim test to fail. Under low lat scan window uses time reservation which would cause any advertisement start to be skipped to next advertising interval. As the fix in this PR corrects the `ticks_window` value, and correct `ticks_slot` is reserved, hence failing the test.

I am thinking on, how to resolve this:
1. not have the msg_mesh_low_lat test
2. make mesh stack tolerant to delayed advertising start

**Comment by PavelVPV** [comment]
> @PavelVPV and @Thalley This fix causes the friendship msg_mesh_low_lat bsim test to fail. Under low lat scan window uses time reservation which would cause any advertisement start to be skipped to next advertising interval. As the fix in this PR corrects the `ticks_window` value, and correct `ticks_slot` is reserved, hence failing the test.
> 
> I am thinking on, how to resolve this:
> 
> 1. not have the msg_mesh_low_lat test
> 2. make mesh stack tolerant to delayed advertising start

Could cherry-pick this patch: https://github.com/PavelVPV/zephyr/commit/939896b6df243e33304bf8ac173df8af39b5811d ?

**Comment by cvinayak** [comment]
> Could cherry-pick this patch: [PavelVPV@939896b](https://github.com/PavelVPV/zephyr/commit/939896b6df243e33304bf8ac173df8af39b5811d) ?

Thank you very much, it helped.

**Review by PavelVPV — APPROVED** [review]

**Review by Thalley — APPROVED** [review]

**Pull request merged**